lib/WPA/Andersen.cpp

```diff
@@ -106,6 +129,15 @@ void AndersenBase::analyze()
         // Start solving constraints
         DBOUT(DGENERAL, outs() << SVFUtil::pasMsg("Start Solving Constraints\n"));
 
+        bool ourAlarm = !callersAlarm();
+        // If it's not our alarm, it's probably FS's alarm and we don't want to mess with that.
+        // TODO: maybe make AnderTimeLimit and FsTimeLimit work together?
+        if (ourAlarm && Options::AnderTimeLimit != 0)
+        {
+            signal(SIGALRM, &timeLimitReached);
+            alarm(Options::AnderTimeLimit);
+        }
+
```

Comment:
The code here is a bit more inside the solver and can we make this as a method and return a bool value or some other ways to shorten and hidden this?  It would also be good to explain `ourAlarm` 's meaning.

Reply:
Good point, I've abstracted it all into a couple of methods.